trite-osprey/src/List.js

```diff
@@ -0,0 +1,219 @@
+import React from 'react';
+import Cart from './Cart';
+
+class List extends React.Component{
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      beers : [],
+      currentBeer: null,
+      searchString: '',
+      visible: false,
+      active: this.props.active || 0,
+      beerList: [],
+      totalPrice: 0,
+      currentBeerObjects: []
+    }
+    this.handleClick = this.handleClick.bind(this)
+  }
+
+  componentWillMount() {
+    this.getAllBeers()
+  }
+
+  getAllBeers() {
+    fetch('http://localhost:5000/beers', {
+      method: 'get',
+    })
+    .then(response => response.json())
+    .then(results => {
+      this.setState({
+        beers: results
+      })
+    })
+  }
+
+  handleChange(event) {
+    this.setState( { searchString: event.target.value} )
+  }
+
+  hide() {
+    this.setState({visible: false})
+  }
+
+  handleClick(event, index) {
+    this.setState({visible: true, active: index, currentBeer: this.state.beers[index] })
```

Comment:
It seems weird to me that you need to set the index in your state more than once. If `this.state.active` is already sent to the index you want, than you might consider deriving `currentBeer` from `this.state.beers[this.state.active]` within your render function. Putting the same data twice into your state object is a code smell.